api/API_MANIFESTO.adoc

```diff
@@ -43,6 +43,12 @@ By default we don't use other classes than JDK + ours.**
 
 * Be clear about whats PUBLIC API and what internals/implementations are. Place your implementations under `impl.*` package.
 
+# Naming
+
+* Naming things is very important
+
+* Think of the impact that a name has on all the components (documentation, console/CLI, tutorials): an established name, even if it doesn't convey all the meanings, is often better than a new, more precise name.
+
 # Documentation
```

Comment:
I'd like to approach the documentation requirements by content type rather than format. I guess this is more or less what is already here but slightly different:

Concept / This should be one of the first pieces of content imo. With API 2.0 we should explain the fundamental principles behind it to help users make the transition and get the best experience. A new API is a new approach. We should aim to provide a deep explanation about that.

Reference / Javadoc that provides an authoritative source that is consistent, complete, and accurate.

Procedures / This encompasses the simple tutorial and user doc. For the user doc I think the Diataxis model is interesting in how tutorials are differentiated from how to: https://diataxis.fr/how-to-guides/#

When approaching the user doc I think we need to determine what are the "recipes" we should provide in our content set. It would be nice to define the various "how to" procedures that could make up the user doc.